paasta_tools/utils.py

```diff
@@ -2562,6 +2562,13 @@ def get_topology_spread_constraints(self) -> List[Dict[str, Any]]:
         """List of TopologySpreadConstraints that will be applied to all Pods in the cluster"""
         return self.config_dict.get("topology_spread_constraints", [])
 
+    def get_datastore_credentials_vault_overrides(self) -> Optional[str]:
```

Comment:
I *think*
```suggestion
    def get_datastore_credentials_vault_overrides(self) -> Dict[str, str]:
```
might be the right type here? there's no way for this to return `None` and we generally expect a dict of strings to strings, right?

Reply:
not sure what Dict[str, str] implies - would this mean it's always non-empty? I get confused since we could also return an empty dict

Reply:
ah, sorry - Dict[str, str] just means that if the dict is not empty, the keys will always be strings and the key-values will also be strings (afaik, there's no way in mypy to type that a dict will never be empty)